dora/core/common/src/main/java/alluxio/membership/DefaultEtcdMembershipManager.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.membership;
+
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.wire.WorkerInfo;
+import alluxio.wire.WorkerState;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.JsonParseException;
+import io.etcd.jetcd.KeyValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * MembershipManager backed by configured etcd cluster. Only use active users
+ */
+public class DefaultEtcdMembershipManager implements MembershipManager {
```

Comment:
better with ServiceRegistryMembershipManager ?
as ServiceRegistry is the provider for Service Discovery service